python/tank/util/shotgun/publish_resolve.py

```diff
@@ -79,41 +79,88 @@ def resolve_publish_path(tk, sg_publish_data):
+    ``SHOTGUN_PATH_MAC_RENDERS``. The general syntax for this is
+    ``SHOTGUN_PATH_WINDOWS|MAC|LINUX_STORAGENAME``.
+
+    .. note:: If you define a ``SHOTGUN_PATH_MAC_RENDERS`` environment variable
+              and there is a local storage ``Renders`` with a mac path set,
+              the local storage value will be used and a warning will be logged.
+
+    .. note:: If no storage can be resolved for the current operating system,
               a :class:`~sgtk.util.PublishPathNotDefinedError` is raised.
 
+
     **Resolving file urls**
 
-    The method also supports the resolution of ``file://`` urls. Such urls
-    are not multi platform and the local storages and environment variables
-    will therefore be used to try to resolve such paths in case of ambiguity.
+    The method also supports the resolution of ``file://`` urls. Contrary to
+    local file storage objects in Shotgun (see above), these paths are not
+    stored in a multi-os representation but are just defined on the operating
+    system where they were created.
+
+    If you are trying to resolve a ``file://`` url on an os platform different
+    from the one where where the url was created, the method will attempt to
+    resolve it into a valid path using several different approaches:
+
+    - First, it will look for the three environment variables ``SHOTGUN_PATH_WINDOWS``,
+      ``SHOTGUN_PATH_MAC``, and ``SHOTGUN_PATH_LINUX``. If these are defined,
+      the method will attempt to translate the path this way. For example, if
+      you are trying to resolve ``file:///prod/proj_x/assets/bush/file.txt`` on
+      windows, you could set up ``SHOTGUN_PATH_WINDOWS=P:\prod`` and
+      ``SHOTGUN_PATH_LINUX=/prod`` in order to hint the way the path should be
+      resolved.
 
-    - First, local storage settings will be downloaded from Shotgun and added to
-      a path translation table.
+    - If you wanted to use more than one environment variable, this is possible
+      by extending the above syntax with a name, for example ``SHOTGUN_PATH_LINUX_SECONDARY``.
 
-    - Next, any environment variables of the form ``SHOTGUN_PATH_WINDOWS|MAC|LINUX``
-      or ``SHOTGUN_PATH_WINDOWS|MAC|LINUX_STORAGENAME`` will be added
-      to the translation table.
+    - In addition, all existing local storage definitions will be downloaded
+      and used in the same way.
 
-    - The ``file://`` path will be compared against all the existing roots
-      for all operating systems. The first match detected will be used to translate
-      the path into the current operating system platform.
+    - If a local storage has been defined, but an operating system is missing,
+      this can be supplied via an environment variable. For example, if there
+      is a local storage named ``Renders`` that is defined on Linux and Windows,
+      you can extend to support mac by creating an environment variable named
+      ``SHOTGUN_PATH_MAC_RENDERS``. The general syntax for this is
```

Comment:
it's one and the same! you start out with things as env vars, and then over time you can migrate these env vars into local storages in Shotgun, which will then take precedence.